identity/src/main/java/com/android/identity/GattServer.java

```diff
@@ -302,12 +294,30 @@ public void onCharacteristicWriteRequest(BluetoothDevice device,
                                              int offset,
                                              byte[] value) {
         UUID charUuid = characteristic.getUuid();
-        mLog.transport("onCharacteristicWriteRequest: "
+        mLog.transport("onCharacteristicWriteRequest: " + device.getAddress() + " "
                 + characteristic.getUuid() + " " + offset + " " + Util.toHex(value));
+
+        // If we are connected to a device, ignore write from any other device
+        if (mCurrentConnection != null &&
+                !device.getAddress().equals(mCurrentConnection.getAddress())) {
+            Log.e(TAG, "Ignoring characteristic write request from "
+            + device.getAddress() + " since we're already connected to "
+            + mCurrentConnection.getAddress());
+            return;
+        }
+
         if (charUuid.equals(mCharacteristicStateUuid) && value.length == 1) {
             if (value[0] == 0x01) {
                 // Close server socket when the connection was done by state characteristic
                 stopL2CAPServer();
+                if (mCurrentConnection != null) {
```

Comment:
Similar to checking for null here, shouldn't we check for null below in the `mCharacteristicClient2ServerUuid` branch? (line 328).

Seems like a null connection on receiving that UUID means we should ignore that UUID.

Reply:
Makes sense, I'll do that - thanks!